File: apple/shared/Models/Activity.swift
import Foundation

public struct Activity: Identifiable, Hashable, Codable, Sendable {
    public let id: Int
    public let type: String           // "olympiad" or "textbook"
    public let subject: String
    public let date: String
    public let sortKey: String
    public let name: String
    public let country: String?
    public let finished: Int
    public let highlighted: Int

    enum CodingKeys: String, CodingKey {
        case id, type, subject, date, name, country, finished, highlighted
        case sortKey = "sort_key"
    }

    public var isOlympiad: Bool { type == "olympiad" }
    public var isTextbook: Bool { type == "textbook" }
}

public struct ActivityList: Codable, Sendable {
    public let items: [Activity]
}

File: android/core/src/main/kotlin/com/vivianweidai/science/core/model/Activity.kt
package com.vivianweidai.science.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Activity(
    val id: Int,
    val type: String,
    val subject: String,
    val date: String,
    @SerialName("sort_key") val sortKey: String,
    val name: String,
    val highlighted: Int,
    val subjects: List<String>? = null,
    val invited: Int? = null,
    val borderline: Int? = null,
    val competitive: Int? = null,
) {
    val isOlympiad: Boolean get() = type == "olympiad"
    val isTextbook: Boolean get() = type == "textbook"
}

@Serializable
data class ActivityList(val items: List<Activity>)
